agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -482,179 +455,221 @@ protected static void replace_instructions(
 
   /**
-   * Reads the file that specifies method calls that should be replaced by other method calls. The
-   * file is of the form:
-   *
-   * <p>[regex] [orig-method-def] [new-method-name]
+   * Reads the replacement file specifying method calls that should be replaced by other method
+   * calls. The file may have lines of the form:
    *
-   * <p>where the orig_method-def is of the form:
+   * <p>[original-name] [replacement-name]
    *
-   * <p>fully-qualified-method-name (args)
+   * <p>where both names are either method names, class names or package names.
    *
-   * <p>Blank lines and // comments are ignored. The orig-method-def is replaced by a call to
-   * new-method-name with the same arguments in any classfile that matches the regular expressions.
-   * All method names and argument types should be fully-qualified.
+   * <p>Blank lines and // comments are ignored. A call to a method in a class of the
+   * orig-package-name is replaced by a call to the same method within the new-package-name. All
+   * method names and argument types should be fully-qualified.
    *
-   * @param map_file the file with map of method substitutions
-   * @throws IOException if file has missing regex
+   * @param in the {@code Reader} for the replacement file
+   * @throws IOException if there is an error reading from the file
    */
-  public void read_map_file(File map_file) throws IOException {
-
-    LineNumberReader lr = new LineNumberReader(new FileReader(map_file));
-    MapFileErrorHandler mfeh = new MapFileErrorHandler(lr, map_file);
-    Pattern current_regex = null;
-    Map<MethodDef, ReplacementClass> map = new LinkedHashMap<>();
+  void readMapFile(Reader in) throws IOException {
+    LineNumberReader lr = new LineNumberReader(in);
     for (String line = lr.readLine(); line != null; line = lr.readLine()) {
-      line = line.replaceFirst("//.*$", "");
-      if (line.trim().length() == 0) continue;
-      if (line.startsWith(" ")) {
-        if (current_regex == null)
-          throw new IOException("No current class regex on line " + lr.getLineNumber());
-        StrTok st = new StrTok(line, mfeh);
-        st.stok.wordChars('.', '.'); // make '.' a word constituent
-        MethodDef md = parse_method(st);
-        String replacement_method_class = st.need_word(); // words including '.'
-        map.put(md, new ReplacementClass(replacement_method_class));
+      line = line.replaceFirst("//.*$", "").trim();
+      if (line.length() == 0) {
+        continue;
+      }
+      Matcher sigMatcher = SIGNATURE_PATTERN.matcher(line);
+      if (sigMatcher.find() && sigMatcher.groupCount() == 2) {
+        String[] arguments = new String[0];
+        String argString = sigMatcher.group(2);
+        if (!argString.isEmpty()) {
+          arguments = argString.split(",");
+        }
+        MethodDef orig = getMethod(sigMatcher.group(1).trim(), arguments);
+        if (sigMatcher.find() && sigMatcher.groupCount() == 2) {
```

Comment:
Why is this (and 7 lines above) using `find()` instead of `matches()` or an anchored pattern?  This code is confusing and error-prone:  it will match a line with two occurrences of SIGNATURE_PATTERN even if there is other junk on the line.  It would be better to create a single regular expression that will match the entire line, and force it to match the entire line.